lib/main.js

```diff
@@ -741,6 +741,19 @@ function GetState(state) {
     MML.SUPER.ID = ID = 0;
     MathJax.OutputJax.CommonHTML.ID = 0;
   }
+
+  // Clear any existing user defined macros, then load macros from state
+  Object.keys(TEX.Definitions.macros).forEach(function(macroName){
+    if (TEX.Definitions.macros[macroName].isUser) {
+      delete TEX.Definitions.macros[macroName];
+    }
+  });
+  if (state && state.macros) {
+    for (var macroName in state.macros) {
+      TEX.Definitions.macros[macroName] = state.macros[macroName];
+      TEX.Definitions.macros[macroName].isUser = true;
```

Comment:
I figured if someone were to serialize the state to JSON, the `isUser` property would be lost because the macro is an array.

If that's not something you want to support, I'm happy to remove it. It's not important for our use case at the moment.